org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormatterSettings.java

```diff
@@ -0,0 +1,131 @@
+
+import org.eclipse.lsp4j.FormattingOptions;
+
+/**
+ * This class is the root of all formatting settings.
+ * It is necessary to update this class for any new additions.
+ * 
+ * All defaults should be set here to eventually be overriden if needed.
+ */
+public class XMLFormatterSettings {
+  private static final Logger LOGGER = Logger.getLogger(XMLFormatterSettings.class.getName());
+
+  //All possible keys
+  public final static String TAB_SIZE = "tabSize";
+  public final static String INSERT_SPACES = "insertSpaces";
+  public final static String SPLIT_ATTRIBUTES = "splitAttributes";
+  public final static String JOIN_CDATA_LINES = "joinCDATALines";
+  public final static String FORMAT_COMMENTS = "formatComments";
+  public final static String JOIN_COMMENT_LINES = "joinCommentLines";
+
+  private static int defaultTabSize = 4;
+  private static boolean defaultIndentSpaces = false;
+  private Map<String, Object> settingsMap;
+
+
+  public XMLFormatterSettings() {
+    initializeDefaultSettings();
+  }
+
+  public XMLFormatterSettings(int tabSize, boolean indentSpaces) {
+    this.defaultTabSize = tabSize;
+    this.defaultIndentSpaces = indentSpaces;
+    initializeDefaultSettings();
+  }
+
+  public XMLFormatterSettings(FormattingOptions formatterOptions) {
+    updateFromFormattingOptions(formatterOptions);
+    initializeDefaultSettings();
+  }
+
+  //***Keep this up to date with all possible settings***
+  public XMLFormatterSettings initializeDefaultSettings() { 
+    this.settingsMap = new ConcurrentHashMap<>();
+    
+    //TAB_SIZE and INSERT_SPACES are eventually provided by didChangeConfigurations or FormattingOptions
+
+    this.settingsMap.put(SPLIT_ATTRIBUTES, false);
+    this.settingsMap.put(JOIN_CDATA_LINES, false);
+    this.settingsMap.put(FORMAT_COMMENTS, true);
+    this.settingsMap.put(JOIN_COMMENT_LINES, false);
+    return this;
+  }
+
+  public void updateSetting(String key, Object value) {
+    if(key != null && value != null) {
+      settingsMap.put(key, value);
+    }
+  }
+
+  public void updateSettingsFromMap(Map<String, Object> newSettings) {
+    if(newSettings != null) {
+      for (String key : newSettings.keySet()) {
+        Object value = newSettings.get(key);
+        if(value instanceof Double) {
+              value = ((Double) value).intValue();
+        }
+        this.settingsMap.put(key, value);
+      }
+    }
+  }
+
+  public void updateFromFormattingOptions(FormattingOptions formattingOptions) {
+    if(settingsMap.get(TAB_SIZE) == null) {
+      int tabSize = formattingOptions.getTabSize();
+      updateSetting(XMLFormatterSettings.TAB_SIZE, tabSize);
+    }
+		if(settingsMap.get(INSERT_SPACES) == null) {
+      boolean indentSpaces = formattingOptions.isInsertSpaces();
+		  updateSetting(XMLFormatterSettings.INSERT_SPACES, indentSpaces);
+    }
+	}
+
+
+  public boolean getSplitAttributes() {
+    return (boolean) getMapObject(SPLIT_ATTRIBUTES);
+  }
+
+  public boolean getJoinCDATALines() {
```

Comment:
Rename to isJoinCDATALines (generally you use is* if it returns boolean and get*if it returns Boolean)